docs/architecture/log_levels.md

```diff
@@ -0,0 +1,122 @@
+| LevelWarn | Error performing action, can proceed |
+| LevelError | Error performing action, cannot proceed |
+| LevelReportedError | Actor exited unexpectedly |
+| LevelReportedError | Actor panicked |
+
+## Reference
+
+### Other log level definitions
+
+#### OpenTelemetry
+
+| SeverityNumber range | Range name | Meaning |
+|---|---|---|
+| 1-4   | TRACE | A fine-grained debugging event. Typically disabled in default configurations. |
+| 5-8   | DEBUG | A debugging event. |
+| 9-12  | INFO  | An informational event. Indicates that an event happened. |
+| 13-16 | WARN  | A warning event. Not an error but is likely more important than an informational event. |
+| 17-20 | ERROR | An error event. Something went wrong. |
+| 21-24 | FATAL | A fatal error such as application or system crash. |
+
+[Source](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber).
+
+#### Golang (slog)
+
+```golang
+// Names for common levels.
+//
+// Level numbers are inherently arbitrary,
+// but we picked them to satisfy three constraints.
+// Any system can map them to another numbering scheme if it wishes.
+//
+// First, we wanted the default level to be Info, Since Levels are ints, Info is
+// the default value for int, zero.
+//
+// Second, we wanted to make it easy to use levels to specify logger verbosity.
+// Since a larger level means a more severe event, a logger that accepts events
+// with smaller (or more negative) level means a more verbose logger. Logger
+// verbosity is thus the negation of event severity, and the default verbosity
+// of 0 accepts all events at least as severe as INFO.
+//
+// Third, we wanted some room between levels to accommodate schemes with named
+// levels between ours. For example, Google Cloud Logging defines a Notice level
+// between Info and Warn. Since there are only a few of these intermediate
+// levels, the gap between the numbers need not be large. Our gap of 4 matches
+// OpenTelemetry's mapping. Subtracting 9 from an OpenTelemetry level in the
+// DEBUG, INFO, WARN and ERROR ranges converts it to the corresponding slog
+// Level range. OpenTelemetry also has the names TRACE and FATAL, which slog
+// does not. But those OpenTelemetry levels can still be represented as slog
+// Levels by using the appropriate integers.
+const (
+	LevelDebug Level = -4
+	LevelInfo  Level = 0
+	LevelWarn  Level = 4
+	LevelError Level = 8
+)
+```
+
+#### GCP
+
+| Level | Meaning |
+|---|---|
+| DEFAULT 	(0) 	| The log entry has no assigned severity level. |
+| DEBUG 	(100)	| Debug or trace information. |
+| INFO 		(200)	| Routine information, such as ongoing status or performance. |
+| NOTICE 	(300)	| Normal but significant events, such as start up, shut down, or a configuration change. |
+| WARNING 	(400)	| Warning events might cause problems. |
+| ERROR 	(500)	| Error events are likely to cause problems. |
+| CRITICAL 	(600)	| Critical events cause more severe problems or outages. |
```

Comment:
huh.. _this_ highlites what feels off to me.

You're kinda defining `error` and `critical`, but `critical` is getting conflated with the reporting path. 

So maybe we want to use these, and then decide, separately, what level to report on?

Reply:
I'm not really sure we need a `critical`?

I have been thinking of `LevelReportedError` as basically "error, and also we want to put this error into the reporting path" -- that's why we re-level `LevelReportedError` to `LevelError` when shipping logs.